pkg/cli/admin/upgrade/upgrade.go

```diff
@@ -307,15 +314,12 @@ func (o *Options) Run() error {
 		return nil
 
 	default:
-		if c := findCondition(cv.Status.Conditions, configv1.OperatorDegraded); c != nil && c.Status == configv1.ConditionTrue {
-			prefix := "No upgrade is possible due to an error"
-			if c := findCondition(cv.Status.Conditions, configv1.OperatorProgressing); c != nil && c.Status == configv1.ConditionTrue && len(c.Message) > 0 {
-				prefix = c.Message
+		if c := findCondition(cv.Status.Conditions, ClusterStatusFailing); c != nil {
+			if c.Status != configv1.ConditionFalse {
+				fmt.Fprintf(o.Out, "%s=%s:\n\n  Reason: %s\n  Message: %s\n\n", c.Type, c.Status, c.Reason, c.Message)
 			}
-			if len(c.Message) > 0 {
-				return fmt.Errorf("%s:\n\n  Reason: %s\n  Message: %s\n\n", prefix, c.Reason, c.Message)
-			}
-			return fmt.Errorf("The cluster can't be upgraded, see `oc describe clusterversion`")
+		} else {
+			fmt.Fprintf(o.ErrOut, "warning: No current %s info, see `oc describe clusterversion` for more details", ClusterStatusFailing)
```

Comment:
Why is this a warning? Also it should be s/`No current %s info`/ `No information on current %s`/

Reply:
It's a warning, because we should always be setting `Failing`.  Ideally to `Failing=False`, but sometimes to `Failing=True`.  A ClusterVersion with no `Failing` at all is a sign that something is probably wrong with the CVO.

`...No current Failing info, see...` sounds pretty clear to me.  I'd be fine rephrasing to something like `...No current conditions with type Failing, see...` here and below for `Progressing` if you prefer.